### PR TITLE
Create custom exercises

### PR DESCRIPTION
**Pull request by ctperry0301**
Added Ability to Create Custom Exercises.

### CODE REVIEW

File: actions/exerciseActions.js [hunk 1]
```diff
@@ -7,6 +7,15 @@ const initalizeExercises = (exercises) => {
    }
 }
 
+const addCustomExercise = (exercises) => {
+  return {
+     type: ADD_CUSTOM_EXERCISE,
+     exercises,
```
Comment (jpoist97) [inline actions/exerciseActions.js]: nit: should just be exercise right?

File: components/AddExercises/CreateCustomExercise.jsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+import RNPickerSelect from 'react-native-picker-select';
+import * as firebase from 'firebase';
+import actions from '../../actions/index';
+
+const Name = styled.Text`
+font-family: 'Montserrat_600SemiBold';
+font-size: 16px;
+margin-left: 8px;
+margin-bottom: 10px;
+padding: 4px;
```
Comment (jpoist97) [inline components/AddExercises/CreateCustomExercise.jsx]: Style: Could you tab these over a once, they don't get picked up by eslint for some reason

File: components/AddExercises/CreateCustomExercise.jsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    const customExerciseRef = firebase.firestore()
+      .collection('users')
+      .doc(currentUser)
+      .collection('customExercises');
+    const ExerciseDoc = await customExerciseRef.add(exercise);
+    dispatch(actions.exercises.addCustomExercise({
+      muscleGroups: muscleGroup,
+      name: workoutName,
+      id: ExerciseDoc.id,
+    }));
```
Comment (jpoist97) [inline components/AddExercises/CreateCustomExercise.jsx]: Could you move most of this code into an action that uses thunk, you can pretty much just copy this code over. Also could you add the creating exercise doc as a function in api.js

**Review by jpoist97 — COMMENTED** [review]
Mostly nitpicks, make sure I correctly fixed the merge conflicts and this should be good to go in.

File: components/AddExercises/CreateCustomExercise.jsx (hunk 3 above)
Comment (ctperry0301) [reply components/AddExercises/CreateCustomExercise.jsx]: Im not quite sure how to do this, is there a good, easy, example somewhere?

Comment (jpoist97) [reply components/AddExercises/CreateCustomExercise.jsx]: Yeah I think a good example is in workoutActions, the updateWorkout function.

Basically you just have the action return a function that takes in dispatch as the only parameter. From there you can do whatever you want in the function and then dispatch an action to the reducer. So all you'd do in your action creator is call the database stuff from api and then dispatch the action that your current actionCreator already dispatches

**Review by jpoist97 — APPROVED** [review]

**Pull request merged**